src/System.Numerics.Vectors/tests/Performance/Vector2/Normalize.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Numerics.Tests
+{
+    public static partial class Perf_Vector2
+    {
+        [Benchmark(InnerIterationCount = VectorTests.DefaultInnerIterationsCount)]
+        public static void NormalizeBenchmark()
+        {
+            var expectedResult = new Vector2(-16777216.0f, 16777216.0f);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                Vector2 result;
+
+                using (iteration.StartMeasurement())
+                {
+                    result = NormalizeTest();
+                }
+
+                VectorTests.AssertEqual(expectedResult, result);
+            }
+        }
+
+        public static Vector2 NormalizeTest()
+        {
+            var result = VectorTests.Vector2Value;
+
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                result += Vector2.Normalize(result);
```

Comment:
That is to say, we need to be changing the inputs and consuming the outputs; otherwise, as the JIT compiler is improved, we will end up optimizing it away (as was already the case for some of the tests).

When that happens, someone who isn't aware of the current limitations of the perf tests will just see a perf gain and not realize that the tests are passing but 'broken' and we will be back in the same state as before this PR.